### PR TITLE
Replace pkg_resources with pkgutil.

### PR DESCRIPTION
**Pull request by benjaminp**
pkg_resources (part of setuptools) is overkill for reading resource files. The standard library module pkgutil can do that just fine.

### CODE REVIEW

**Comment by lidizheng** [comment]
Hi, Benjamin. Thank you for submitting the PR. I have a few thoughts about this problem. As you said, the building process of the project depends on `setuptools` and `pkg_resources` came for free. Also, `pkg_resources` are used through out the project in many places, it is better to use one package consistently. Can you provide stronger reasoning to migrate to `pkgutil`?

File: src/python/grpcio_tests/tests/interop/resources.py [hunk 1]
```diff
@@ -14,27 +14,25 @@
...
 
 _ROOT_CERTIFICATES_RESOURCE_PATH = 'credentials/ca.pem'
 _PRIVATE_KEY_RESOURCE_PATH = 'credentials/server1.key'
 _CERTIFICATE_CHAIN_RESOURCE_PATH = 'credentials/server1.pem'
 
 
 def test_root_certificates():
-    return pkg_resources.resource_string(__name__,
-                                         _ROOT_CERTIFICATES_RESOURCE_PATH)
+    return pkgutil.get_data('tests.interop', _ROOT_CERTIFICATES_RESOURCE_PATH)
```
Comment (lidizheng) [inline src/python/grpcio_tests/tests/interop/resources.py]: Using the `__name__` seems better than hard coding the module name.

**Comment by benjaminp** [comment]
> Hi, Benjamin. Thank you for submitting the PR. I have a few thoughts about this problem. As you said, the building process of the project depends on `setuptools` and `pkg_resources` came for free. Also, `pkg_resources` are used through out the project in many places, it is better to use one package consistently. Can you provide stronger reasoning to migrate to `pkgutil`?

1. Just because `setuptools` is required in the build environment doesn't mean it needs to be required in the runtime environment. It should be possible to install a `grpcio` wheel into any environment which has just the packages in `install_requires`

2. `pkg_resources` does a huge amount of IO at import-time. Importing it is the death of any program that needs a quick startup time.

File: src/python/grpcio_tests/tests/interop/resources.py (hunk 1 above)
Comment (benjaminp) [reply src/python/grpcio_tests/tests/interop/resources.py]: I've now switched back to `__name__`.

**Comment by grpc-testing** [comment]
```
Objective-C binary sizes
*****************STATIC******************
  New size                      Old size
 2,015,578      Total (=)      2,015,578

 No significant differences in binary sizes

***************FRAMEWORKS****************
  New size                      Old size
11,116,191      Total (<)     11,116,193

 No significant differences in binary sizes


```

File: src/python/grpcio/grpc/_cython/_cygrpc/security.pyx.pxi [hunk 2]
```diff
@@ -14,14 +14,14 @@
...
+import pkgutil
 
 
 cdef grpc_ssl_roots_override_result ssl_roots_override_callback(
     char **pem_root_certs) nogil:
   with gil:
-    temporary_pem_root_certs = pkg_resources.resource_string(
-        __name__.rstrip('.cygrpc'), '_credentials/roots.pem')
+    temporary_pem_root_certs = pkgutil.get_data(
+        __name__[:-len('.cygrpc')], '_credentials/roots.pem')
```
Comment (lidizheng) [inline src/python/grpcio/grpc/_cython/_cygrpc/security.pyx.pxi]: Can we change back to `rstrip` here? These two is not always equivalent when the `__name__` not ending with `.cygrpc`.

Comment (benjaminp) [reply src/python/grpcio/grpc/_cython/_cygrpc/security.pyx.pxi]: `rstrip` is the wrong method to use here; the argument is [not a suffix](https://docs.python.org/3/library/stdtypes.html#str.rstrip). My latest update changes the code to do what was probably originally intended, though.

**Comment by benjaminp** [comment]
Latest update applies yapf.

**Comment by grpc-testing** [comment]
```
Objective-C binary sizes
*****************STATIC******************
  New size                      Old size
 2,015,578      Total (=)      2,015,578

 No significant differences in binary sizes

***************FRAMEWORKS****************
  New size                      Old size
11,116,199      Total (=)     11,116,199

 No significant differences in binary sizes


```

**Comment by benjaminp** [comment]
Thank you for the review, lidizheng! Are you also the right person to look at #16916?

**Pull request merged**